ivadomed/loader/files_dataset.py

```diff
@@ -0,0 +1,239 @@
+        """Load the json file and return the dictionary"""
+        # Given a JSON file, try to load the file pairing from it.
+        assert loader_json.get(DataloaderKW.TYPE) == "FILES"
+
+        self.path_data = loader_json.get(DataloaderKW.PATH_DATA, ".")
+
+        if DataloaderKW.IMAGE_GROUND_TRUTH in loader_json.keys():
+            self.filename_pairs = self.validate_update_filename_pairs(
+                loader_json,
+                self.n_expected_gt,
+                self.n_expected_input,
+                self.path_data
+            )
+
+        return self.path_data, self.filename_pairs
+
+    def validate_update_filename_pairs(self,
+                                       loader_json: dict,
+                                       n_expected_gt: int,
+                                       n_expected_input: int,
+                                       path_data: Path) -> list:
+        """
+        Validate the JSON file and update the filename_pairs after ensuring no extra or insufficient files presents.
+
+        Args:
+            loader_json:
+            n_expected_gt:
+            n_expected_input:
+            path_data:
+
+        Returns:
+
+        """
+        filename_pairs = []
+
+        if not loader_json.get(DataloaderKW.IMAGE_GROUND_TRUTH):
+            raise KeyError("Expected V2 loader configuration files but the Image/Groundtruth key was not found in the loader JSON.")
+
+        list_image_ground_truth_pairs: list = loader_json.get(DataloaderKW.IMAGE_GROUND_TRUTH)
+        list_image_ground_truth_pairs_filtered: list = []
+        # Go through each subject
+        for a_subject_image_ground_truth_pair in list_image_ground_truth_pairs:
+
+            skip_subject_flag = False
+
+            # 2 lists: Subject List + Ground Truth list
+            assert len(a_subject_image_ground_truth_pair) == 2
+
+            # Validate and trim Subject files list
+            list_subject_specific_images: list = a_subject_image_ground_truth_pair[0]
+            if len(list_subject_specific_images) > n_expected_input:
+                list_subject_specific_images = list_subject_specific_images[:n_expected_input]
+
+            # Validate and trim Ground Truth files list
+            list_subject_specific_gt: list = a_subject_image_ground_truth_pair[1]
+            if len(list_subject_specific_gt) > n_expected_gt:
+                list_subject_specific_gt = list_subject_specific_gt[:n_expected_gt]
+
+            # Go check every file and if any of them don't exist, skip the subject
+            for subject_images_index, path_file in enumerate(
+                    list_subject_specific_images
+            ):
+                list_subject_specific_images[subject_images_index] = ensure_absolute_path(
+                    path_file, path_data
+                )
+                if not list_subject_specific_images[subject_images_index]:
+                    skip_subject_flag = True
```

Comment:
I was thinking to push this above i.e before trimming the lists bcz it may happen, an edge case, there are 3 inputs given but while slicing first 2 are taken, and one of them doesn't exist and the one rejected while slicing exists then we might unnecessarily be rejecting the subject...

Also, I would like to maintain a list of subjects or filename pairs being rejected and then display all of them in the end as a warning rather than each one (Todo: check again how does it come across!)

Reply:
Todo: add a test for the edge case I have in mind as mentioned above